x/market/keeper/swap_test.go

```diff
@@ -75,13 +75,13 @@ func TestComputeInternalSwap(t *testing.T) {
 
 	for i := 0; i < 100; i++ {
 		offerCoin := sdk.NewDecCoin(core.MicroSDRDenom, lunaPriceInSDR.MulInt64(rand.Int63()+1).TruncateInt())
-		retCoin, err := input.MarketKeeper.ComputeInternalSwap(input.Ctx, offerCoin, core.MicroLunaDenom)
+		retCoin, err := input.MarketKeeper.ComputeInternalSwap(input.Ctx, offerCoin, core.MicroLunaDenom, false)
```

Comment:
See what I mean? We don't need the additional argument here to distinguish between sim/production if you checked the total supply directly in the message handler instead of the swap computation routines.